### PR TITLE
+angstrom.0.16.0

### PR DESCRIPTION
**Pull request by seliopou**

### CODE REVIEW

File: packages/angstrom-async/angstrom-async.0.16.0/opam [hunk 1]
```diff
@@ -0,0 +1,23 @@
+opam-version: "2.0"
+maintainer: "Spiros Eliopoulos <spiros@inhabitedtype.com>"
+authors: [ "Spiros Eliopoulos <spiros@inhabitedtype.com>" ]
+license: "BSD-3-clause"
+homepage: "https://github.com/inhabitedtype/angstrom"
+bug-reports: "https://github.com/inhabitedtype/angstrom/issues"
+dev-repo: "git+https://github.com/inhabitedtype/angstrom.git"
+build: [
+  ["dune" "subst"] {pinned}
```
Comment (haochenx) [inline packages/angstrom-async/angstrom-async.0.16.0/opam:9]:
```suggestion
  ["dune" "subst"] {dev}
```
`pinned` for `subst` is not correct and is a bug fixed in [dune.2.7.0](https://github.com/ocaml/dune/blob/main/CHANGES.md#270-13082020)

File: packages/angstrom-lwt-unix/angstrom-lwt-unix.0.16.0/opam [hunk 2]
```diff
@@ -0,0 +1,24 @@
+opam-version: "2.0"
+maintainer: "Spiros Eliopoulos <spiros@inhabitedtype.com>"
+authors: [ "Spiros Eliopoulos <spiros@inhabitedtype.com>" ]
+license: "BSD-3-clause"
+homepage: "https://github.com/inhabitedtype/angstrom"
+bug-reports: "https://github.com/inhabitedtype/angstrom/issues"
+dev-repo: "git+https://github.com/inhabitedtype/angstrom.git"
+build: [
+  ["dune" "subst"] {pinned}
```
Comment (haochenx) [inline packages/angstrom-lwt-unix/angstrom-lwt-unix.0.16.0/opam:9]:
```suggestion
  ["dune" "subst"] {dev}
```
`pinned` for `subst` is not correct and is a bug fixed in [dune.2.7.0](https://github.com/ocaml/dune/blob/main/CHANGES.md#270-13082020)

File: packages/angstrom-unix/angstrom-unix.0.16.0/opam [hunk 3]
```diff
@@ -0,0 +1,23 @@
+opam-version: "2.0"
+maintainer: "Spiros Eliopoulos <spiros@inhabitedtype.com>"
+authors: [ "Spiros Eliopoulos <spiros@inhabitedtype.com>" ]
+license: "BSD-3-clause"
+homepage: "https://github.com/inhabitedtype/angstrom"
+bug-reports: "https://github.com/inhabitedtype/angstrom/issues"
+dev-repo: "git+https://github.com/inhabitedtype/angstrom.git"
+build: [
+  ["dune" "subst"] {pinned}
```
Comment (haochenx) [inline packages/angstrom-unix/angstrom-unix.0.16.0/opam:9]:
```suggestion
  ["dune" "subst"] {dev}
```
`pinned` for `subst` is not correct and is a bug fixed in [dune.2.7.0](https://github.com/ocaml/dune/blob/main/CHANGES.md#270-13082020)

File: packages/angstrom/angstrom.0.16.0/opam [hunk 4]
```diff
@@ -0,0 +1,33 @@
+opam-version: "2.0"
+maintainer: "Spiros Eliopoulos <spiros@inhabitedtype.com>"
+authors: [ "Spiros Eliopoulos <spiros@inhabitedtype.com>" ]
+license: "BSD-3-clause"
+homepage: "https://github.com/inhabitedtype/angstrom"
+bug-reports: "https://github.com/inhabitedtype/angstrom/issues"
+dev-repo: "git+https://github.com/inhabitedtype/angstrom.git"
+build: [
+  ["dune" "subst"] {pinned}
```
Comment (haochenx) [inline packages/angstrom/angstrom.0.16.0/opam:9]:
```suggestion
  ["dune" "subst"] {dev}
```
`pinned` for `subst` is not correct and is a bug fixed in [dune.2.7.0](https://github.com/ocaml/dune/blob/main/CHANGES.md#270-13082020)

**Review by haochenx — COMMENTED** [review]
Please also take a look at the failing lower-bounds checks, e.g.

(angstrom-async)
```
#=== ERROR while compiling angstrom-async.0.16.0 ==============================#
# context              2.2.0~alpha3 | linux/x86_64 | ocaml-base-compiler.4.08.1 | pinned(https://github.com/inhabitedtype/angstrom/archive/0.16.0.tar.gz)
# path                 ~/.opam/4.08/.opam-switch/build/angstrom-async.0.16.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p angstrom-async -j 71
# exit-code            1
# env-file             ~/.opam/log/angstrom-async-7-694e91.env
# output-file          ~/.opam/log/angstrom-async-7-694e91.out
### output ###
#       ocamlc async/.angstrom_async.objs/byte/angstrom_async.{cmo,cmt} (exit 2)
# (cd _build/default && /home/opam/.opam/4.08/bin/ocamlc.opt -w -40 -safe-string -g -bin-annot -I async/.angstrom_async.objs/byte -I /home/opam/.opam/4.08/lib/angstrom -I /home/opam/.opam/4.08/lib/async -I /home/opam/.opam/4.08/lib/async_extra -I /home/opam/.opam/4.08/lib/async_kernel -I /home/opam/.opam/4.08/lib/async_kernel/persistent_connection_kernel -I /home/opam/.opam/4.08/lib/async_rpc_kernel -I /home/opam/.opam/4.08/lib/async_unix -I /home/opam/.opam/4.08/lib/async_unix/thread_pool -I /home/opam/.opam/4.08/lib/async_unix/thread_safe_ivar -I /home/opam/.opam/4.08/lib/base -I /home/opam/.opam/4.08/lib/base/caml -I /home/opam/.opam/4.08/lib/base/md5 -I /home/opam/.opam/4.08/lib/base/shadow_stdlib -I /home/opam/.opam/4.08/lib/base_bigstring -I /home/opam/.opam/4.08/lib/base_quickcheck -I /home/opam/.opam/4.08/lib/bin_prot -I /home/opam/.opam/4.08/lib/bin_prot/shape -I /home/opam/.opam/4.08/lib/core -I /home/opam/.opam/4.08/lib/core/nano_mutex -I /home/opam/.opam/4.08/lib/core/squeue -I /home/opam/.opam/4.08/lib/core/uuid -I /home/opam/.opam/4.08/lib/core_kernel -I /home/opam/.opam/4.08/lib/core_kernel/base_for_tests -I /home/opam/.opam/4.08/lib/core_kernel/moption -I /home/opam/.opam/4.08/lib/core_kernel/thread_pool_cpu_affinity -I /home/opam/.opam/4.08/lib/core_kernel/thread_safe_queue -I /home/opam/.opam/4.08/lib/core_kernel/uopt -I /home/opam/.opam/4.08/lib/core_kernel/uuid -I /home/opam/.opam/4.08/lib/fieldslib -I /home/opam/.opam/4.08/lib/jane-street-headers -I /home/opam/.opam/4.08/lib/ocaml/threads -I /home/opam/.opam/4.08/lib/parsexp -I /home/opam/.opam/4.08/lib/ppx_assert/runtime-lib -I /home/opam/.opam/4.08/lib/ppx_bench/runtime-lib -I /home/opam/.opam/4.08/lib/ppx_compare/runtime-lib -I /home/opam/.opam/4.08/lib/ppx_enumerate/runtime-lib -I /home/opam/.opam/4.08/lib/ppx_expect/collector -I /home/opam/.opam/4.08/lib/ppx_expect/common -I /home/opam/.opam/4.08/lib/ppx_expect/config -I /home/opam/.opam/4.08/lib/ppx_hash/runtime-lib -I /home/opam/.opam/4.08/lib/ppx_inline_test/config -I /home/opam/.opam/4.08/lib/ppx_inline_test/runtime-lib -I /home/opam/.opam/4.08/lib/ppx_module_timer/runtime -I /home/opam/.opam/4.08/lib/ppx_sexp_conv/runtime-lib -I /home/opam/.opam/4.08/lib/protocol_version_header -I /home/opam/.opam/4.08/lib/result -I /home/opam/.opam/4.08/lib/sexplib -I /home/opam/.opam/4.08/lib/sexplib/unix -I /home/opam/.opam/4.08/lib/sexplib0 -I /home/opam/.opam/4.08/lib/spawn -I /home/opam/.opam/4.08/lib/splittable_random -I /home/opam/.opam/4.08/lib/stdio -I /home/opam/.opam/4.08/lib/time_now -I /home/opam/.opam/4.08/lib/typerep -I /home/opam/.opam/4.08/lib/variantslib -intf-suffix .ml -no-alias-deps -o async/.angstrom_async.objs/byte/angstrom_async.cmo -c -impl async/angstrom_async.ml)
# File "async/angstrom_async.ml", line 52, characters 14-24:
# 52 |     finalize (p.continue bigstring ~off:0 ~len:(String.length s) Complete) `Eof
#                    ^^^^^^^^^^
# Error: This function has type
#          Angstrom.bigstring ->
#          Angstrom.Unbuffered.more -> 'a Angstrom.Unbuffered.state
#        It is applied to too many arguments; maybe you forgot a `;'.
#     ocamlopt async/.angstrom_async.objs/native/angstrom_async.{cmx,o} (exit 2)
# (cd _build/default && /home/opam/.opam/4.08/bin/ocamlopt.opt -w -40 -safe-string -g -I async/.angstrom_async.objs/byte -I async/.angstrom_async.objs/native -I /home/opam/.opam/4.08/lib/angstrom -I /home/opam/.opam/4.08/lib/async -I /home/opam/.opam/4.08/lib/async_extra -I /home/opam/.opam/4.08/lib/async_kernel -I /home/opam/.opam/4.08/lib/async_kernel/persistent_connection_kernel -I /home/opam/.opam/4.08/lib/async_rpc_kernel -I /home/opam/.opam/4.08/lib/async_unix -I /home/opam/.opam/4.08/lib/async_unix/thread_pool -I /home/opam/.opam/4.08/lib/async_unix/thread_safe_ivar -I /home/opam/.opam/4.08/lib/base -I /home/opam/.opam/4.08/lib/base/caml -I /home/opam/.opam/4.08/lib/base/md5 -I /home/opam/.opam/4.08/lib/base/shadow_stdlib -I /home/opam/.opam/4.08/lib/base_bigstring -I /home/opam/.opam/4.08/lib/base_quickcheck -I /home/opam/.opam/4.08/lib/bin_prot -I /home/opam/.opam/4.08/lib/bin_prot/shape -I /home/opam/.opam/4.08/lib/core -I /home/opam/.opam/4.08/lib/core/nano_mutex -I /home/opam/.opam/4.08/lib/core/squeue -I /home/opam/.opam/4.08/lib/core/uuid -I /home/opam/.opam/4.08/lib/core_kernel -I /home/opam/.opam/4.08/lib/core_kernel/base_for_tests -I /home/opam/.opam/4.08/lib/core_kernel/moption -I /home/opam/.opam/4.08/lib/core_kernel/thread_pool_cpu_affinity -I /home/opam/.opam/4.08/lib/core_kernel/thread_safe_queue -I /home/opam/.opam/4.08/lib/core_kernel/uopt -I /home/opam/.opam/4.08/lib/core_kernel/uuid -I /home/opam/.opam/4.08/lib/fieldslib -I /home/opam/.opam/4.08/lib/jane-street-headers -I /home/opam/.opam/4.08/lib/ocaml/threads -I /home/opam/.opam/4.08/lib/parsexp -I /home/opam/.opam/4.08/lib/ppx_assert/runtime-lib -I /home/opam/.opam/4.08/lib/ppx_bench/runtime-lib -I /home/opam/.opam/4.08/lib/ppx_compare/runtime-lib -I /home/opam/.opam/4.08/lib/ppx_enumerate/runtime-lib -I /home/opam/.opam/4.08/lib/ppx_expect/collector -I /home/opam/.opam/4.08/lib/ppx_expect/common -I /home/opam/.opam/4.08/lib/ppx_expect/config -I /home/opam/.opam/4.08/lib/ppx_hash/runtime-lib -I /home/opam/.opam/4.08/lib/ppx_inline_test/config -I /home/opam/.opam/4.08/lib/ppx_inline_test/runtime-lib -I /home/opam/.opam/4.08/lib/ppx_module_timer/runtime -I /home/opam/.opam/4.08/lib/ppx_sexp_conv/runtime-lib -I /home/opam/.opam/4.08/lib/protocol_version_header -I /home/opam/.opam/4.08/lib/result -I /home/opam/.opam/4.08/lib/sexplib -I /home/opam/.opam/4.08/lib/sexplib/unix -I /home/opam/.opam/4.08/lib/sexplib0 -I /home/opam/.opam/4.08/lib/spawn -I /home/opam/.opam/4.08/lib/splittable_random -I /home/opam/.opam/4.08/lib/stdio -I /home/opam/.opam/4.08/lib/time_now -I /home/opam/.opam/4.08/lib/typerep -I /home/opam/.opam/4.08/lib/variantslib -intf-suffix .ml -no-alias-deps -o async/.angstrom_async.objs/native/angstrom_async.cmx -c -impl async/angstrom_async.ml)
# File "async/angstrom_async.ml", line 52, characters 14-24:
# 52 |     finalize (p.continue bigstring ~off:0 ~len:(String.length s) Complete) `Eof
#                    ^^^^^^^^^^
# Error: This function has type
#          Angstrom.bigstring ->
#          Angstrom.Unbuffered.more -> 'a Angstrom.Unbuffered.state
#        It is applied to too many arguments; maybe you forgot a `;'.
```

(angstrom-lwt-unix)
```
[ERROR] The compilation of cstruct.2.4.1 failed at "make".

#=== ERROR while compiling cstruct.2.4.1 ======================================#
# context              2.2.0~alpha3 | linux/x86_64 | ocaml-base-compiler.4.06.1 | file:///home/opam/opam-repository
# path                 ~/.opam/4.06/.opam-switch/build/cstruct.2.4.1
# command              ~/.opam/opam-init/hooks/sandbox.sh build make
# exit-code            2
# env-file             ~/.opam/log/cstruct-7-af8af0.env
# output-file          ~/.opam/log/cstruct-7-af8af0.out
### output ###
# ocaml setup.ml -build 
# ocamlfind ocamlopt unix.cmxa -I /home/opam/.opam/4.06/lib/ocamlbuild /home/opam/.opam/4.06/lib/ocamlbuild/ocamlbuildlib.cmxa -linkpkg myocamlbuild.ml /home/opam/.opam/4.06/lib/ocamlbuild/ocamlbuild.cmx -o myocamlbuild
# /home/opam/.opam/4.06/bin/ocamlfind ocamlc -g -ccopt -O3 -ccopt -Wall -package sexplib -package ocplib-endian.bigstring -package ocplib-endian -package bytes -package bigarray -c lib/cstruct_stubs.c
# mv cstruct_stubs.o lib/cstruct_stubs.o
# /home/opam/.opam/4.06/bin/ocamlfind ocamlmklib -o lib/cstruct_stubs lib/cstruct_stubs.o
# /home/opam/.opam/4.06/bin/ocamlfind ocamldep -package sexplib -package ocplib-endian.bigstring -package ocplib-endian -package bytes -package bigarray -modules lib/cstruct.mli > lib/cstruct.mli.depends
# /home/opam/.opam/4.06/bin/ocamlfind ocamlc -c -g -annot -bin-annot -principal -ccopt -O3 -ccopt -Wall -package sexplib -package ocplib-endian.bigstring -package ocplib-endian -package bytes -package bigarray -I lib -o lib/cstruct.cmi lib/cstruct.mli
# /home/opam/.opam/4.06/bin/ocamlfind ocamldep -package sexplib -package ocplib-endian.bigstring -package ocplib-endian -package bytes -package bigarray -modules lib/cstruct.ml > lib/cstruct.ml.depends
# /home/opam/.opam/4.06/bin/ocamlfind ocamlc -c -g -annot -bin-annot -principal -ccopt -O3 -ccopt -Wall -package sexplib -package ocplib-endian.bigstring -package ocplib-endian -package bytes -package bigarray -I lib -o lib/cstruct.cmo lib/cstruct.ml
# + /home/opam/.opam/4.06/bin/ocamlfind ocamlc -c -g -annot -bin-annot -principal -ccopt -O3 -ccopt -Wall -package sexplib -package ocplib-endian.bigstring -package ocplib-endian -package bytes -package bigarray -I lib -o lib/cstruct.cmo lib/cstruct.ml
# File "lib/cstruct.ml", line 154, characters 0-138:
# Warning 3: deprecated: [@@noalloc] should be used instead of "noalloc"
# File "lib/cstruct.ml", line 156, characters 0-132:
# Warning 3: deprecated: [@@noalloc] should be used instead of "noalloc"
# File "lib/cstruct.ml", line 158, characters 0-132:
# Warning 3: deprecated: [@@noalloc] should be used instead of "noalloc"
# File "lib/cstruct.ml", line 160, characters 0-132:
# Warning 3: deprecated: [@@noalloc] should be used instead of "noalloc"
# File "lib/cstruct.ml", line 162, characters 0-132:
# Warning 3: deprecated: [@@noalloc] should be used instead of "noalloc"
# File "lib/cstruct.ml", line 164, characters 0-117:
# Warning 3: deprecated: [@@noalloc] should be used instead of "noalloc"
# File "lib/cstruct.ml", line 166, characters 0-102:
# Warning 3: deprecated: [@@noalloc] should be used instead of "noalloc"
# /home/opam/.opam/4.06/bin/ocamlfind ocamlc -a -dllib -lcstruct_stubs -cclib -lcstruct_stubs lib/cstruct.cmo -o lib/cstruct.cma
# /home/opam/.opam/4.06/bin/ocamlfind ocamlopt -c -g -annot -bin-annot -principal -ccopt -O3 -ccopt -Wall -package sexplib -package ocplib-endian.bigstring -package ocplib-endian -package bytes -package bigarray -I lib -o lib/cstruct.cmx lib/cstruct.ml
# + /home/opam/.opam/4.06/bin/ocamlfind ocamlopt -c -g -annot -bin-annot -principal -ccopt -O3 -ccopt -Wall -package sexplib -package ocplib-endian.bigstring -package ocplib-endian -package bytes -package bigarray -I lib -o lib/cstruct.cmx lib/cstruct.ml
# File "lib/cstruct.ml", line 154, characters 0-138:
# Warning 3: deprecated: [@@noalloc] should be used instead of "noalloc"
# File "lib/cstruct.ml", line 156, characters 0-132:
# Warning 3: deprecated: [@@noalloc] should be used instead of "noalloc"
# File "lib/cstruct.ml", line 158, characters 0-132:
# Warning 3: deprecated: [@@noalloc] should be used instead of "noalloc"
# File "lib/cstruct.ml", line 160, characters 0-132:
# Warning 3: deprecated: [@@noalloc] should be used instead of "noalloc"
# File "lib/cstruct.ml", line 162, characters 0-132:
# Warning 3: deprecated: [@@noalloc] should be used instead of "noalloc"
# File "lib/cstruct.ml", line 164, characters 0-117:
# Warning 3: deprecated: [@@noalloc] should be used instead of "noalloc"
# File "lib/cstruct.ml", line 166, characters 0-102:
# Warning 3: deprecated: [@@noalloc] should be used instead of "noalloc"
# /home/opam/.opam/4.06/bin/ocamlfind ocamlopt -a -cclib -lcstruct_stubs lib/cstruct.cmx -o lib/cstruct.cmxa
# /home/opam/.opam/4.06/bin/ocamlfind ocamlopt -shared -I lib/ lib/cstruct.cmxa lib/cstruct.cmx -o lib/cstruct.cmxs
# /home/opam/.opam/4.06/bin/ocamlfind ocamldep -package sexplib -package ocplib-endian.bigstring -package ocplib-endian -package bytes -package bigarray -package lwt.unix -modules lwt/lwt_cstruct.mli > lwt/lwt_cstruct.mli.depends
# /home/opam/.opam/4.06/bin/ocamlfind ocamlc -c -g -annot -bin-annot -principal -I lib -package sexplib -package ocplib-endian.bigstring -package ocplib-endian -package bytes -package bigarray -package lwt.unix -I lwt -I lib -o lwt/lwt_cstruct.cmi lwt/lwt_cstruct.mli
# + /home/opam/.opam/4.06/bin/ocamlfind ocamlc -c -g -annot -bin-annot -principal -I lib -package sexplib -package ocplib-endian.bigstring -package ocplib-endian -package bytes -package bigarray -package lwt.unix -I lwt -I lib -o lwt/lwt_cstruct.cmi lwt/lwt_cstruct.mli
# ocamlfind: Error from package `threads': Missing -thread or -vmthread switch
# Command exited with code 2.
# E: Failure("Command ''/home/opam/.opam/4.06/bin/ocamlbuild' lib/libcstruct_stubs.a lib/dllcstruct_stubs.so lib/cstruct.cma lib/cstruct.cmxa lib/cstruct.a lib/cstruct.cmxs lwt/lwt_cstruct.cma lwt/lwt_cstruct.cmxa lwt/lwt_cstruct.a lwt/lwt_cstruct.cmxs unix/unix_cstruct.cma unix/unix_cstruct.cmxa unix/unix_cstruct.a unix/unix_cstruct.cmxs ppx/ppx_cstruct.cmx ppx/ppx_cstruct.cmo ppx/standalone.native -use-ocamlfind -tag debug' terminated with error code 10")
# make: *** [Makefile:7: build] Error 1
```

**Comment by haochenx** [comment]
This PR also (rightfully?) removes dependency to `result`, which seems to cause a few revdeps (which is not this PR's fault in principle: they should have declared dependency on result themselves.)

We probably want to patch those packages as well (but probably not in this PR.)

File: packages/angstrom-async/angstrom-async.0.16.0/opam [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+dev-repo: "git+https://github.com/inhabitedtype/angstrom.git"
+build: [
+  ["dune" "subst"] {dev}
+  ["dune" "build" "-p" name "-j" jobs]
+  ["dune" "runtest" "-p" name "-j" jobs] {with-test}
+]
+depends: [
+  "ocaml" {>= "4.04.1"}
+  "dune" {>= "1.8"}
+  "angstrom" {>= "0.7.0"}
```
Comment (mseri) [inline packages/angstrom-async/angstrom-async.0.16.0/opam:16]:
```suggestion
  "angstrom" {>= "0.9.0"}
```
Should fix
```
# File "async/angstrom_async.ml", line 52, characters 14-24:
# 52 |     finalize (p.continue bigstring ~off:0 ~len:(String.length s) Complete) `Eof
#                    ^^^^^^^^^^
# Error: This function has type
#          Angstrom.bigstring ->
#          Angstrom.Unbuffered.more -> 'a Angstrom.Unbuffered.state
#        It is applied to too many arguments; maybe you forgot a `;'.
#     ocamlopt async/.angstrom_async.objs/native/angstrom_async.{cmx,o} (exit 2)
# (cd _build/default && /home/opam/.opam/4.14/bin/ocamlopt.opt -w -40 -safe-string -g -I async/.angstrom_async.objs/byte -I async/.angstrom_async.objs/native -I /home/opam/.opam/4.14/lib/angstrom -I /home/opam/.opam/4.14/lib/async -I /home/opam/.opam/4.14/lib/async/async_command -I /home/opam/.opam/4.14/lib/async/async_quickcheck -I /home/opam/.opam/4.14/lib/async/async_rpc -I /home/opam/.opam/4.14/lib/async_kernel -I /home/opam/.opam/4.14/lib/async_kernel/persistent_connection_kernel -I /home/opam/.opam/4.14/lib/async_rpc_kernel -I /home/opam/.opam/4.14/lib/async_unix -I /home/opam/.opam/4.14/lib/async_unix/thread_pool -I /home/opam/.opam/4.14/lib/async_unix/thread_safe_ivar -I /home/opam/.opam/4.14/lib/base -I /home/opam/.opam/4.14/lib/base/base_internalhash_types -I /home/opam/.opam/4.14/lib/base/caml -I /home/opam/.opam/4.14/lib/base/md5 -I /home/opam/.opam/4.14/lib/base/shadow_stdlib -I /home/opam/.opam/4.14/lib/base_bigstring -I /home/opam/.opam/4.14/lib/base_quickcheck -I /home/opam/.opam/4.14/lib/bin_prot -I /home/opam/.opam/4.14/lib/bin_prot/shape -I /home/opam/.opam/4.14/lib/core -I /home/opam/.opam/4.14/lib/core/bigstring_unix -I /home/opam/.opam/4.14/lib/core/error_checking_mutex -I /home/opam/.opam/4.14/lib/core/iobuf_unix -I /home/opam/.opam/4.14/lib/core/linux_ext -I /home/opam/.opam/4.14/lib/core/nano_mutex -I /home/opam/.opam/4.14/lib/core/squeue -I /home/opam/.opam/4.14/lib/core/time_stamp_counter -I /home/opam/.opam/4.14/lib/core/uuid -I /home/opam/.opam/4.14/lib/core_kernel -I /home/opam/.opam/4.14/lib/core_kernel/base_for_tests -I /home/opam/.opam/4.14/lib/core_kernel/bounded_int_table -I /home/opam/.opam/4.14/lib/core_kernel/caml_unix -I /home/opam/.opam/4.14/lib/core_kernel/flags -I /home/opam/.opam/4.14/lib/core_kernel/iobuf -I /home/opam/.opam/4.14/lib/core_kernel/moption -I /home/opam/.opam/4.14/lib/core_kernel/pairing_heap -I /home/opam/.opam/4.14/lib/core_kernel/sexp_hidden_in_test -I /home/opam/.opam/4.14/lib/core_kernel/thread_pool_cpu_affinity -I /home/opam/.opam/4.14/lib/core_kernel/thread_safe_queue -I /home/opam/.opam/4.14/lib/core_kernel/timing_wheel -I /home/opam/.opam/4.14/lib/core_kernel/tuple_pool -I /home/opam/.opam/4.14/lib/core_kernel/uopt -I /home/opam/.opam/4.14/lib/core_kernel/uuid -I /home/opam/.opam/4.14/lib/core_kernel/version_util -I /home/opam/.opam/4.14/lib/fieldslib -I /home/opam/.opam/4.14/lib/jane-street-headers -I /home/opam/.opam/4.14/lib/ocaml/threads -I /home/opam/.opam/4.14/lib/parsexp -I /home/opam/.opam/4.14/lib/ppx_assert/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_bench/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_compare/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_enumerate/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_expect/collector -I /home/opam/.opam/4.14/lib/ppx_expect/common -I /home/opam/.opam/4.14/lib/ppx_expect/config -I /home/opam/.opam/4.14/lib/ppx_expect/config_types -I /home/opam/.opam/4.14/lib/ppx_hash/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_here/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_inline_test/config -I /home/opam/.opam/4.14/lib/ppx_inline_test/runtime-lib -I /home/opam/.opam/4.14/lib/ppx_module_timer/runtime -I /home/opam/.opam/4.14/lib/ppx_sexp_conv/runtime-lib -I /home/opam/.opam/4.14/lib/protocol_version_header -I /home/opam/.opam/4.14/lib/result -I /home/opam/.opam/4.14/lib/sexplib -I /home/opam/.opam/4.14/lib/sexplib/unix -I /home/opam/.opam/4.14/lib/sexplib0 -I /home/opam/.opam/4.14/lib/spawn -I /home/opam/.opam/4.14/lib/splittable_random -I /home/opam/.opam/4.14/lib/stdio -I /home/opam/.opam/4.14/lib/time_now -I /home/opam/.opam/4.14/lib/timezone -I /home/opam/.opam/4.14/lib/typerep -I /home/opam/.opam/4.14/lib/variantslib -intf-suffix .ml -no-alias-deps -o async/.angstrom_async.objs/native/angstrom_async.cmx -c -impl async/angstrom_async.ml)
# File "async/angstrom_async.ml", line 52, characters 14-24:
# 52 |     finalize (p.continue bigstring ~off:0 ~len:(String.length s) Complete) `Eof
#                    ^^^^^^^^^^
# Error: This function has type
#          Angstrom.bigstring ->
#          Angstrom.Unbuffered.more -> 'a Angstrom.Unbuffered.state
#        It is applied to too many arguments; maybe you forgot a `;'.
```

**Comment by mseri** [comment]
The lower bounds are fixed, the rest of the failures are unrelated. @seliopou can you update the opam file for angstrom-async in your repo?

**Pull request merged**